src/Illuminate/Database/Eloquent/Concerns/HasEvents.php

```diff
@@ -190,6 +190,29 @@ protected function fireModelEvent($event, $halt = true)
         );
     }
 
+    protected function fireModelBeforeEvent($event)
+    {
+        $eventResponses = $this->fireModelEvent($event, false);
+
+        if ($eventResponses === false) {
+            return false;
+        }
+
+        if ($eventResponses === null) {
+            return null;
+        }
+
+        $eventResponses = array_map(function ($response) {
```

Comment:
Wouldn't this be simpler to understand? Some other parts of the framework use this approach:

~~~php
foreach (Arr::wrap($eventResponses) as $response) {
    if ($response === false) {
        return false;
    }
}

return true;
~~~

Some examples:

https://github.com/laravel/framework/blob/49368f4f50e63c543511bad983e7bcef038e9794/src/Illuminate/Auth/SessionGuard.php#L442-L451

https://github.com/laravel/framework/blob/49368f4f50e63c543511bad983e7bcef038e9794/src/Illuminate/Collections/Collection.php#L545-L556